### DISCUSSION TITLE
Which IDE can I use to debug this project

### DISCUSSION THREAD

**Discussion by pdlogingithub** [Beep6581/RawTherapee#6706] [Q&A]
Can I use visual studio to debug and set break point on this project?if not, which IDE should I use, Is there any tutorial on this suject?
thank you.

**Comment by Lawrence37** [comment]
RawTherapee is a fairly straightforward C++ application, so any C++ capable debugger should work. How to use the debugger depends on which debugger you decide to use. Just search for guides and tutorials for the debugger and make sure you are using a debug build of RawTherapee for the best experience. RawPedia mentions using GDB which in my experience works well. Any IDE that integrates with GDB should be good.

**Comment by pdlogingithub** [comment]
Thank you for the reply.
Does RawTherapee have a log file or log window so that I can print some message and see them?

**Reply by Lawrence37** [reply]
No, all logs are sent to stdout and stderr via cout, cerr, printf, etc.. In Windows, I have had problems seeing the output in the console, but redirection to a file works.
